### PR TITLE
RFC: New SQL planner framework

### PR DESCRIPTION
**Pull request by leiysky**
I hereby agree to the terms of the CLA available at: https://databend.rs/policies/cla/

## Summary

Add RFC about #1217 

## Changelog

- Documentation

### CODE REVIEW

**Comment by leiysky** [comment]
@BohuTANG @sundy-li I've just written a RFC about refactoring SQL planner, please feel free to comment.

File: website/databend/docs/rfcs/query/0005-new-sql-planner-framework.md [hunk 1]
```diff
@@ -0,0 +1,313 @@
...
+
+```
+postgres=# select a from t t1, t t2;
+ERROR:  column reference "a" is ambiguous
+LINE 1: select a from t t1, t t2;
+```
+
+While you can reference the column with a canonical name like `t.a` since the table name is required to be unique in a query context.
+
+Currently, `databend` uses `DataSchema` to represent input and output relation schema, which can not provide enough information to handle the case shown above. In a `DataSchema`, each column is represented with `DataField`, which has following definition:
```
Comment (BohuTANG) [inline website/databend/docs/rfcs/query/0005-new-sql-planner-framework.md:45]: 👍

**Comment by BohuTANG** [comment]
Great work, thank you @leiysky 
This RFC very clear and understandable

**Comment by zhang2014** [comment]
Everything is perfect. thank you @leiysky

File: website/databend/docs/rfcs/query/0005-new-sql-planner-framework.md [hunk 2]
```diff
@@ -0,0 +1,313 @@
+- Start date: 2021/09/13
+- Tracking issues: https://github.com/datafuselabs/datafuse/issues/1217
```
Comment (PsiACE) [inline website/databend/docs/rfcs/query/0005-new-sql-planner-framework.md:2]: datafuse -> databend

Comment (leiysky) [reply website/databend/docs/rfcs/query/0005-new-sql-planner-framework.md]: Fixed.

Thanks for pointing this out.

**Review by BohuTANG — APPROVED** [review]
LGTM

**Pull request merged**